### PR TITLE
Auth client

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: core/kazoo/src/kz_maps.erl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+%% ====================================================================
+%% API functions
+%% ====================================================================
+-export([merge/2]).
+-export([get/2, get/3]).
+-export([keys_to_atoms/1, keys_to_atoms/2]).
+
+-spec merge(map(), map()) -> map().
+merge(Map1, Map1) -> Map1;
```
Comment (jamesaimonetti) [inline core/kazoo/src/kz_maps.erl]: Can you add unit tests so we can see the intended usages/results? I think there may be more efficient ways to do this merge but I wouldn't want to break functionality, so tests would be great to make sure there isn't a regression.

Comment (lazedo) [inline core/kazoo/src/kz_maps.erl]: @jamesaimonetti added the test

**Comment by jamesaimonetti** [comment]
@lazedo i rebased and added perf testing for the merge functions. Want to try locally `make perf` in `core/kazoo` to make sure you see similar speed differences?

File: core/kazoo/src/kz_maps.erl [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+%% ====================================================================
+-export([merge/2
+        ,merge_r/2
+        ,merge_l/2
+        ]).
+-export([get/2, get/3]).
+-export([keys_to_atoms/1, keys_to_atoms/2]).
+
+-spec merge(map(), map()) -> map().
+merge(Map1, Map1) -> Map1;
```
Comment (fenollp) [inline core/kazoo/src/kz_maps.erl]: Can you add an is_map(Map1) on this line?

**Pull request merged**